### ISSUE TITLE
Clarification regarding the TripletSemiHardLoss

### ISSUE THREAD

**Issue by sayakpaul** [tensorflow/addons#1754]
Hi,

I am referring to this tutorial: https://www.tensorflow.org/addons/tutorials/losses_triplet. It uses the `TripletSemiHardLoss`. The network architecture used in the tutorial is as follows:

```
model = tf.keras.Sequential([
    tf.keras.layers.Conv2D(filters=64, kernel_size=2, padding='same', activation='relu', input_shape=(28,28,1)),
    tf.keras.layers.MaxPooling2D(pool_size=2),
    tf.keras.layers.Dropout(0.3),
    tf.keras.layers.Conv2D(filters=32, kernel_size=2, padding='same', activation='relu'),
    tf.keras.layers.MaxPooling2D(pool_size=2),
    tf.keras.layers.Dropout(0.3),
    tf.keras.layers.Flatten(),
    tf.keras.layers.Dense(256, activation=None), # No activation on final dense layer
    tf.keras.layers.Lambda(lambda x: tf.math.l2_normalize(x, axis=1)) # L2 normalize embeddings

])
```

From `model.summary()` I get - 

```
Layer (type)                 Output Shape              Param #   
=================================================================
conv2d_10 (Conv2D)           (None, 28, 28, 64)        320       
_________________________________________________________________
max_pooling2d_10 (MaxPooling (None, 14, 14, 64)        0         
_________________________________________________________________
dropout_10 (Dropout)         (None, 14, 14, 64)        0         
_________________________________________________________________
conv2d_11 (Conv2D)           (None, 14, 14, 32)        8224      
_________________________________________________________________
max_pooling2d_11 (MaxPooling (None, 7, 7, 32)          0         
_________________________________________________________________
dropout_11 (Dropout)         (None, 7, 7, 32)          0         
_________________________________________________________________
flatten_2 (Flatten)          (None, 1568)              0         
_________________________________________________________________
dense_10 (Dense)             (None, 256)               401664    
_________________________________________________________________
lambda_5 (Lambda)            (None, 256)               0         
=================================================================
Total params: 410,208
Trainable params: 410,208
Non-trainable params: 0
```

I see the output is a 256-d vector. At this point, I am confused as to how `TripletSemiHardLoss` is actually calculating the loss because while calling `fit`, `train_dataset` is fed that yields images and labels. If you could shed some lights on this, that would be great.
